README.md

```diff
@@ -36,7 +36,9 @@ Here's an example of how to write an upgrade test:
 ```go
 pt := pulumitest.NewPulumiTest(t, "path-to-a-pulumi-program-dir",
   // Use our local implementation for the new version
-  opttest.AttachProviderServer("my-provider-name", myProviderServerFactory))
+  opttest.AttachProviderServer("my-provider-name", func(pt pulumitest.PulumiTest) (pulumirpc.ResourceProviderServer, error) {
+    // Return the implementation of your provider
```

Comment:
Maybe we could link to an example here? It would be great to know what the test expects in terms of this ResourceProviderServer.